### ISSUE TITLE
tapir zio stops immediately after starting on version 1.2.3 - works fine on 1.2.2

### ISSUE THREAD

**Issue by softedgedk** [softwaremill/tapir#2615]
After bumping to version 1.2.3, my tapir with zio http stops immediately after startup; confirmed by downgrading to 1.2.2 and undoing Server.start <=> Server.serve changes + adding/removing .provide(ServerConfig.live(), Server.live) (+ fixing imports zhttp.http.HttpApp  <=> zio.http.HttpApp and zhttp.service.Server <=> zio.http.Server

**Comment by adamw** [comment]
I think there was an update to ZIO 2.0.4 (https://github.com/softwaremill/tapir/releases/tag/v1.2.3). Do you have the right version of ZIO? Are you logging all errors, also during startup - maybe there are some that are thrown when the application gets set up?

If it's not something that is local, I'd need a reproducing example, plus some more information - which ZIO version, how are you using it, etc.

**Comment by softedgedk** [comment]
Confirmed - forcing zio 2.0.2 makes it work again:

dependencyOverrides ++= Seq(
    "dev.zio" %% "zio" % "2.0.2",
    "dev.zio" %% "zio-streams" % "2.0.2"
)

**Comment by softedgedk** [comment]
(and before it was zio version 2.0.4 - also forcing version 2.0.3 does not work)

**Comment by softedgedk** [comment]
Could this be captured in this error: https://github.com/softwaremill/tapir/issues/2605

**Comment by adamw** [comment]
@softedgedk no, that's a zio1 issue, and the failure is intermittent

I would guess that there is some ClassCastException, CNFE or similar that happens during startup. Do you log all fiber failures, and do you log uncaught exceptions on startup? Without a reproducing example, I don't think I can do much. The example server in our codebase works just fine.

**Comment by argast** [comment]
I'm getting following error when using sttp client with zio >= `2.0.3`:

```[error] Exception in thread "zio-fiber-8,5,1,0" java.lang.Error: Defect in zio.ZEnvironment: Set(SttpBackend[=λ %A → ZIO[-Any,+Throwable,+A],+{package::WebSockets & ZioStreams}]) statically known to be contained within the environment are missing
[error]         Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1"
[error]                 Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1"
[error]                         Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1"
[error] Exception in thread "zio-fiber-6,1,0" java.lang.Error: Defect in zio.ZEnvironment: Set(SttpBackend[=λ %A → ZIO[-Any,+Throwable,+A],+{package::WebSockets & ZioStreams}]) statically known to be contained within the environment are missing
```

It might be related as it works fine with zio overrides above.

**Comment by adamw** [comment]
I'm not sure how zio's compatibility looks like, but I'd try sticking to sttp, tapir & zio versions which use exactly the same zio version (e.g. 2.0.3).

If there are problems after upgrading to 2.0.4, I'd report this to zio, as it's a problem with compatibility, not tapir/sttp

**Comment by frekw** [comment]
This was a regression in 2.0.4, it's fixed in 2.0.5.

**Comment by adamw** [comment]
Closing as it was a problem with ZIO
